src/nodes.coffee

```diff
@@ -232,18 +232,20 @@ exports.Base = class Base
   cacheToCodeFragments: (cacheValues) ->
     [fragmentsToText(cacheValues[0]), fragmentsToText(cacheValues[1])]
 
-  # Construct a node that returns the current node's result.
+  # Construct a node that returns the current node’s result.
   # Note that this is overridden for smarter behavior for
-  # many statement nodes (e.g. If, For)...
-  makeReturn: ({accumulator, mark} = {}) ->
+  # many statement nodes (e.g. `If`, `For`).
+  makeReturn: (results, mark) ->
     if mark
+      # Mark this node as implicitly returned, so that it can be part of the
+      # node metadata returned in the AST.
       @canBeReturned = yes
       return
-    me = @unwrapAll()
-    if accumulator
-      new Call new Literal("#{accumulator}.push"), [me]
+    node = @unwrapAll()
+    if results
+      new Call new Literal("#{results}.push"), [node]
     else
-      new Return me
+      new Return node
```

Comment:
I did, however, rename the variables to be a bit more consistent with the rest of the codebase. I went with `results` because that’s the name of the accumulator variable (such as that returned by an implicit `while` or `for`).